.github/workflows/ci.yml

```diff
@@ -0,0 +1,58 @@
+name: Build and Deploy Documentation using MkDocs
+
+# Controls when the action will run. Triggers the workflow on push events on any branch 
+# but pull request events only for the main branch
+on:
+  push: #  for any branch
+    # branches:
+    #   - '**'
+  pull_request:
+    branches: [ main ]
+  
+  # Allows you to run this workflow manually from the Actions tab
+  workflow_dispatch:
+
+jobs:
+  setup: 
+    name: Setup Ubuntu 
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        python-version: [ 3.8 ]
+
+    steps:
+      - name: Checkout Main
+        uses: actions/checkout@v2
+
+      - name: Set up Python ${{ matrix.python-version }}
+        uses: actions/setup-python@v2
+        with:
+          python-version: ${{ matrix.python-version }}
+          architecture: x64
+      - name: Install dependencies and setup git config
```

Comment:
Think we're missing a `run: |` line here, see the failing job in your fork:
https://github.com/Milstein/nerc-docs/actions/runs/1186358520

Reply:
I tested: https://github.com/Milstein/nerc-docs/actions/runs/1187247540 - commenting the push to main branch part looks like this is the desired behavior.